workflows/pipe-common/scripts/mount_storage.py

```diff
@@ -58,6 +59,27 @@ def __init__(self):
     def is_storage_readable(cls, storage):
         return cls.is_permission_set(storage, READ_MASK)
 
+    # Checks that tool with its version for the current run is allowed to mount this storage
+    # or there is no configured restriction for this storage
+    @classmethod
+    def is_storage_available_for_mount(cls, storage, run):
+        if storage.tools_to_mount:
+            return True
+        if run is None:
+            Logger.info('Run info is not present, storage will not be available for mount', task_name=self.task_name)
+            return False
+
+        tool = run["actualDockerImage"]
+        re_result = re.search(r"([^/]+)/([^:]+):?(:?.*)", tool)
+        registry, image, version = re_result.groups()
+        for tool_to_mount in storage.tools_to_mount:
+            if registry == tool_to_mount["registry"] and image == tool_to_mount["image"]:
+                if not v["version"]:
+                    return True
+                else if tool_to_mount["versions"]:
+                    return version and version in [v["version"] for v in tool_to_mount["versions"]]
```

Comment:
Looks like this piece of code is not correct. Let's use something like the following:

```python
tool_versions_to_mount = tool_to_mount.get('versions', [])
return not tool_versions_to_mount or version in [v["version"] for v in tool_versions_to_mount]
```